modules/platform/nuxeo-drive-server/nuxeo-drive-core/src/test/java/org/nuxeo/drive/fixtures/AuditChangeFinderTestSuite.java

```diff
@@ -79,6 +79,31 @@
     @Inject
     protected TrashService trashService;
 
+    /** @since 11.3 */
+    @Test
+    public void testImpactedUser() {
+        nuxeoDriveManager.registerSynchronizationRoot(session.getPrincipal(), folder2, session);
+        txFeature.nextTransaction();
+
+        // We need to save the lower bound id...
+        long replayFrom = lastEventLogId;
+        List<FileSystemItemChange> changes = getChanges();
+        assertEquals(2, changes.size());
+
+        // ...So we can query the same id range again
+        lastEventLogId = replayFrom;
+        NuxeoPrincipalImpl foo = new NuxeoPrincipalImpl("foo");
+        assertEquals(0, getChanges(foo).size());
+
+        // Now we check the found logs:
+        FileSystemItemChange change = changes.get(0);
+        assertEquals(folder2.getId(), change.getDocUuid());
+        assertEquals("rootRegistered", change.getEventId());
+        change = changes.get(1);
+        assertEquals(folder2.getId(), change.getDocUuid());
+        assertEquals("documentCreated", change.getEventId());
```

Comment:
I have definitely missed something because I was pretty sure it worked but well...obviously...
however I'm not sure I understood your point.
The mongo filter linked (https://github.com/nuxeo/nuxeo/pull/4237/files#diff-d4ec341d2c4e8ec9ac9ff5775f2f006bR152) is not equivalent to the SQL clause: https://github.com/nuxeo/nuxeo/blob/master/modules/platform/nuxeo-drive-server/nuxeo-drive-core/src/main/java/org/nuxeo/drive/service/impl/AuditChangeFinder.java#L350

The mongo filter that corresponds to it is https://github.com/nuxeo/nuxeo/pull/4237/files#diff-d4ec341d2c4e8ec9ac9ff5775f2f006bR157-R158 and is always "ANDed" at the end of the top-level filter array: https://github.com/nuxeo/nuxeo/pull/4237/files#diff-d4ec341d2c4e8ec9ac9ff5775f2f006bR161

Also when registering a root, NuxeoDrive events are fired as well, not only when deleting.

Reply:
Replaying the tests it seems to work correctly:
"foo" can't see the logs where impacted user is "Administrator" in both SQL and MongoDB tests
if I do the same query as foo with Administrator, I see the logs again, so it means the bounds are correct.

There is still a big issue with the clustering test going on. And it seems to me the mongo driver has a problem with dates and timestamps handling.

Reply:
Sorry about the links, they must've moved with the PR evolving.

The point is, whatever implementation you take, SQL or Mongo, remove the code that handles the impacted user name in the XXChangefinder and the test will still pass => the test doesn't test what it should.

By debugging, you'll see that the reason why the _foo_ user gets an empty list of changes is not what you think, it's because it simply doesn't have access to `folder2` (no Read ACL). Indeed, the audit entry relevant to this root registration by _Administrator_ is returned by the audit query for _foo_ but is then filtered in `AuditChangeFinder` by the following code:
```
                if (!session.exists(docRef)) {
                    log.debug("Document {} ({}) doesn't exist, not adding entry to the change summary.",
                            entry::getDocPath, () -> docRef);
                    // Deleted or non accessible documents are mapped to
                    // deleted file system items in a separate event: no need to
                    // try to propagate this event.
                    continue;
                }
```
Look at when the `impactedUserName` property is added to the virtual audit event by [NuxeoDriveFileSystemDeletionListener](https://github.com/nuxeo/nuxeo/blob/master/modules/platform/nuxeo-drive-server/nuxeo-drive-core/src/main/java/org/nuxeo/drive/listener/NuxeoDriveFileSystemDeletionListener.java).